resources/2024-06-resource-extensibility-feature-spec.md

```diff
@@ -0,0 +1,452 @@
+                host:
+                    type: 'string'
+                    description: 'Hostname'
+                env:
+                    type: 'string'
+                    description: 'Radius environment'
+                app:
+                    type: 'string'
+                    description: 'Radius application'
+                secrets:
+                    connectionstring:
+                    type: 'string'
+                    description: 'Connection string for PostgreSQL'
+        ```
+
+    1. Amy adds a version for the schema
+
+         ```
+        namespace: 'Applications.Datastores'
+        apiVersion: 'v1'
+        scope: '/planes/radius/Applications.Datastores'
+        resourceTypes:
+            'Applications.Datastores/postgreSQL':
+                name: 'Name of the resource'
+                description: 'PostgreSQL' 
+                properties: 
+                required: ['host', 'env', 'secrets']
+                host:
+                    type: 'string'
+                    description: 'Hostname'
+                env:
+                    type: 'string'
+                    description: 'Radius environment'
+                app:
+                    type: 'string'
+                    description: 'Radius application'
+                secrets:
+                    connectionstring:
+                    type: 'string'
+                    description: 'Connection string for PostgreSQL'
+        ```
+   1. Amy registers the postgreSQL schema to an existing provider in Radius
+    
+        ```bash
+        rad resource-type add -rp Applications --template-kind yaml -e myenv -v v1
+        ```
+        
+        Radius complies the schema and registers the custom resource type Plaid in UCP
+        
+        Question : Will we continue to accept new types like postgreSQL to `Applications` namespace
+   
+#### Iterating and testing the application platform
+
+ As a platform engineer, I need a productive inner loop for development and testing the application platform. Simple and easy way to iterate on changes, test and deploy them
+
+**Solution in Radius**
+
+1. Bicep tooling support to test and deploy the custom resource types - rad CLI support to generate and publish the Bicep types for the custom resources to the registry.
+1. Documentation to guide the users on how to test and deploy the custom resource types in Radius
+
+*User groups and their needs*
+
+There are two types of users that we need to cater to:
+1. Users who have registered schema using typespec schema format
+1. Users who have registered schema using yaml schema format
+
+*User experience*
+
+1. Deb compiles the typespec schema 
+    
+        ```bash
+        tsp compile
+        ```
+        The typespec schema is compiled and validated for any errors and warnings
+
+1. Deb publishes the bicep types to a registry
+    
+        ```bash
+        rad publish-bicep-types --file contoso.tsp --registry myregistry
```

Comment:
Should we be keeping track of versioning here as well? the index.json file would have an extension version when it gets generated so that should match with the API version